File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        min_distance = inf
        n = len(words)
        for i in range(n):
            if words[i] == target:
                d = abs(startIndex - i)
                c = abs(d-n)
                min_distance = min(min_distance, min(d,c))
        if min_distance == inf: return -1
        else: return min_distance

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.cpp
class Solution {
public:
    int closestTarget(vector<string>& words, string target, int startIndex) {
        int n = static_cast<int>(size(words));
        int left = (startIndex-1)%n;
        int right = (startIndex+1)%n;
        if(target == words[startIndex]) return 0;
        int iterator = 1;
        while(true){
            if(words[right]==target || words[left>=0?left:n+left]==target) return iterator;
            if(right == left) break;
            iterator++;
            left = (left-1+n)%n;
            right = (right+1)%n;
        }
        return -1;
    }
};